### PR TITLE
AES MixColumns detection

### LINKED ISSUE #499: AES MixColumns
Apologize for deleting the template, I do not have much time. The idea is to check for the `^ 0x1B` operation typically used in AES's MixColumns operation. It looks as such:

https://en.wikipedia.org/wiki/Rijndael_MixColumns
```
for (c = 0; c < 4; c++) {
        a[c] = r[c];
        /* h is 0xff if the high bit of r[c] is set, 0 otherwise */
        h = (r[c] >> 7) & 1; /* arithmetic right shift, thus shifting in either zeros or ones */
        b[c] = r[c] << 1; /* implicitly removes high bit because b[c] is an 8-bit char, so we xor by 0x1b and not 0x11b in the next line */
        b[c] ^= h * 0x1B; /* Rijndael's Galois field */
    }
```

Sample 1e9fc7f32bd5522dd0222932eb9f1d8bd0a2e132c7b46cfcc622ad97831e6128 has this at VA 0x0040E3DE

```
 LOBYTE(result) = 0;
  for ( i = 1; i < 4; ++i )
  {
    v4 = 2 * *((_BYTE *)&a1 + i + 3);
    *(&a2 + i) = v4;
    if ( *(&a2 + i - 1) < 0 )
      *(&a2 + i) = v4 ^ 0x1B;
  }
  for ( j = 0; j < 4; ++j )
  {
    if ( (((unsigned __int8)a1 >> j) & 1) != 0 )
      LOBYTE(result) = *(&a2 + j) ^ result;
  }
  return result;
```

No AES rules hit for this sample.

### PR DESCRIPTION
**Pull request by mr-tz**
closes #499

### CODE REVIEW

File: data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# generated using capa explorer for IDA Pro
```
Comment (mike-hunhoff) [inline data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml:1]: ❤️

File: data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      # b[c] ^= h * 0x1B;
+      - instruction:
+        - and:
+          - description: Rijndael's Galois field
+          - mnemonic: xor
+          - operand[1].number: 0x1B
+      - optional:
+        - instruction:
+          - and:
+            - description: b[c] = ...
```
Comment (mike-hunhoff) [inline data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml:41]: Can we be a little more descriptive here? I think I understand the intent but I had to think about it for a second 😅

**Review by mike-hunhoff — APPROVED** [review]
Thank you @mr-tz ! One optional comment to address, otherwise LGTM 🚀

File: data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml (hunk 2 above)
Comment (mr-tz) [inline data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml:41]:
```suggestion
            - description: **b[c] =** r[c] << 1; **b[c] ^=** h * 0x1B;
```

Comment (mr-tz) [reply data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml:41]: How about this?

Comment (mr-tz) [reply data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml:41]: @mike-hunhoff, does the above work?

File: data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      # b[c] ^= h * 0x1B;
+      - instruction:
+        - and:
+          - description: Rijndael's Galois field
+          - mnemonic: xor
+          - operand[1].number: 0x1B
+      - optional:
+        - instruction:
+          - and:
+            - description: **b[c] =** r[c] << 1; **b[c] ^=** h * 0x1B;
```
Comment (mr-tz) [inline data-manipulation/encryption/aes/encrypt-data-using-aes-mixcolumns-step.yml:41]:
```suggestion
            - description: "byte assignment, e.g., in: b[c] = r[c] << 1; b[c] ^= h * 0x1B;"
```

**Pull request merged**